### PR TITLE
BITAU-178 Show shared codes on the search screen

### PR DESCRIPTION
**Pull request by ahaisting-livefront**
## 🎟️ Tracking

https://livefront.atlassian.net/browse/BITAU-178

## 📔 Objective

The goal here is to get Shared codes to show up on the search screen.

## 📸 Screenshots

<video src="https://github.com/user-attachments/assets/0f2dddbd-93af-4848-a564-0eb24d8bd661" width="300" />

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/search/ItemSearchScreen.kt [hunk 1]
```diff
@@ -110,47 +99,7 @@ fun ItemSearchScreen(
...
             }
         }
     }
 }
-
-/**
- * Dialogs displayed within the context of the item search screen.
- */
-@Composable
-private fun ItemSearchDialogs(
```
Comment (ahaisting-livefront) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/search/ItemSearchScreen.kt:135]: These dialogs were never actually showing, so I removed them. There was quite a bit from this vm/screen that was impossible to ever show, so I removed it out. The resulting vm is simple and nice!

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/search/VaultVerificationCodeItem.kt [hunk 2]
```diff
@@ -80,19 +80,18 @@ fun VaultVerificationCodeItem(
             verticalArrangement = Arrangement.SpaceEvenly,
             modifier = Modifier.weight(1f),
         ) {
-            issuer?.let {
+            if (!issuer.isNullOrEmpty()) {
```
Comment (ahaisting-livefront) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/search/VaultVerificationCodeItem.kt:83]: This now matches what we use in `ItemListScreen`

File: app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/search/ItemSearchViewModel.kt [hunk 3]
```diff
@@ -38,15 +39,17 @@ class ItemSearchViewModel @Inject constructor(
     initialState = savedStateHandle[KEY_STATE]
         ?: ItemSearchState(
             searchTerm = "",
-            viewState = ItemSearchState.ViewState.Loading,
-            dialogState = null,
+            viewState = ItemSearchState.ViewState.Empty(message = null),
```
Comment (SaintPatrck) [inline app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/search/ItemSearchViewModel.kt:42]: 👍

File: app/src/test/java/com/bitwarden/authenticator/data/authenticator/repository/util/SharedVerificationCodesStateExtensionsTest.kt [hunk 4]
```diff
@@ -18,4 +19,20 @@ class SharedVerificationCodesStateExtensionsTest {
         assertFalse(SharedVerificationCodesState.SyncNotEnabled.isSyncWithBitwardenEnabled)
         assertTrue(SharedVerificationCodesState.Success(emptyList()).isSyncWithBitwardenEnabled)
     }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `itemsOrEmpty should a non empty list only when state is Success `() {
```
Comment (SaintPatrck) [inline app/src/test/java/com/bitwarden/authenticator/data/authenticator/repository/util/SharedVerificationCodesStateExtensionsTest.kt:25]:
```suggestion
    fun `itemsOrEmpty should return a non empty list only when state is Success `() {
```

**Review by SaintPatrck — COMMENTED** [review]
Super minor typo. Looks great otherwise. Love the view model cleanup!

File: app/src/test/java/com/bitwarden/authenticator/data/authenticator/repository/util/SharedVerificationCodesStateExtensionsTest.kt (hunk 4 above)
Comment (ahaisting-livefront) [reply app/src/test/java/com/bitwarden/authenticator/data/authenticator/repository/util/SharedVerificationCodesStateExtensionsTest.kt]: Thank you!

**Review by SaintPatrck — APPROVED** [review]

**Review by SaintPatrck — APPROVED** [review]

**Pull request merged**